engine/src/main/java/org/terasology/world/block/BlockLifecycleEvent.java

```diff
@@ -32,11 +36,46 @@ public BlockLifecycleEvent(TIntList positions, BlockEntityRegistry registry) {
         this.positions = positions;
     }
 
-    public Iterable<Vector3i> getBlockPositions() {
-        return () -> new BlockPositionIterator(positions, registry);
+    @NotNull
+    @Override
+    public Iterator<Vector3ic> iterator() {
+        if (positions.size() < 3) {
+            return Collections.emptyIterator();
+        }
+        return new Iterator<Vector3ic>() {
+            private Vector3i next = new Vector3i(
+                positions.get(0),
+                positions.get(1),
+                positions.get(2));
+            private final Vector3i current = new Vector3i();
+            private int index = 3;
+            @Override
+            public boolean hasNext() {
+                return next != null;
+            }
+
+            @Override
+            public Vector3ic next() {
+                current.set(next);
```

Comment:
This will throw a NPE when there's no `next` element, right? Should we throw a `NoSuchElementException` instead, as suggested in the docs for `Iterator`?